types/queries.go

```diff
@@ -111,6 +113,64 @@ type AllBalancesResponse struct {
 	Amount Coins `json:"amount"`
 }
 
+type IbcQuery struct {
+	PortId       *PortIdQuery       `json:"port_id,omitempty"`
+	ListChannels *ListChannelsQuery `json:"list_channels,omitempty"`
+	Channel      *ChannelQuery      `json:"channel,omitempty"`
+}
+
+type PortIdQuery struct{}
+
+type PortIdResponse struct {
+	PortId string `json:"port_id"`
+}
+
+type ListChannelsQuery struct {
+	// optional argument
+	PortId string `json:"port_id,omitempty"`
```

Comment:
I was 🤔  if we can limit this to the contract's portID but then came to result that a contract may want to query `transfer` port or others it knows